pkg/collector/corechecks/oracle/oracle.go

```diff
@@ -0,0 +1,158 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package oracle
+
+import (
+	"context"
+	"fmt"
+
+	_ "github.com/godror/godror"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/oracle/config"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	go_ora "github.com/sijms/go-ora/v2"
+)
+
+// Check represents one Oracle instance check.
+type Check struct {
+	core.CheckBase
+	config        *config.CheckConfig
+	db            *sqlx.DB
+	hostname      string
+	dbmEnabled    bool
+	agentVersion  string
+	checkInterval float64
+	tags          []string
+}
+
+// Run executes the check.
+func (c *Check) Run() error {
+	/*
+		sender, err := c.GetSender()
+		if err != nil {
+			log.Error("Get sender Run")
+			return err
+		}
+		sender.Gauge("oracle.helloworld", 10, "", nil)
+		sender.Commit()
+	*/
+	if c.db == nil {
+		db, err := c.Connect()
+		if err != nil {
+			c.Teardown()
+			return err
+		}
+		c.db = db
+	}
+
+	if c.hostname == "" {
+		if c.config.InstanceConfig.ReportedHostname != "" {
+			c.hostname = c.config.InstanceConfig.ReportedHostname
+		} else {
+			hostname, err := hostname.Get(context.TODO())
+			if err != nil {
+				return log.Errorf("getting hostname: %v", err)
+			}
+			c.hostname = hostname
+		}
+	}
+
+	if c.dbmEnabled {
+		err := c.SampleSession()
+		if err != nil {
+			return log.Errorf("Sampling session: %v", err)
```

Comment:
This should probably just return the error, right? I assume that the agent will eventually log it but might also use the error to report the check status. 

```suggestion
			return err
```